src/Swift.Bindings/src/Model/TypeSpec.cs

```diff
@@ -0,0 +1,380 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text;
+
+namespace BindingsGeneration;
+
+public abstract class TypeSpec
+{
+	protected TypeSpec(TypeSpecKind kind)
+	{
+		Kind = kind;
+		GenericParameters = new List<TypeSpec>();
+		Attributes = new List<TypeSpecAttribute>();
+	}
+
+	public TypeSpecKind Kind { get; private set; }
+	public List<TypeSpec> GenericParameters { get; private set; }
+	public bool ContainsGenericParameters { get { return GenericParameters.Count != 0; } }
+	public List<TypeSpecAttribute> Attributes { get; private set; }
+	public bool HasAttributes { get { return Attributes.Count != 0; } }
+	public bool IsInOut { get; set; }
+	public bool IsAny { get; set; }
+	public virtual bool IsEmptyTuple { get { return false; } }
```

Comment:
Is it possible to move IsEmptyTuple to TupleTypeSpec?

Reply:
No - this is a convenience method that can be used on any `TypeSpec` without asking if it's a tuple.